src/advanced/JupyterNotebook/index.tsx

```diff
@@ -0,0 +1,327 @@
+    bodyColumnStyle,
+}) => {
+    return (
+        <>
+            <Grid item xs={1}>
+                {idColumnContent}
+            </Grid>
+            <Grid item xs={11}>
+                <Paper variant={'outlined'} style={bodyColumnStyle}>
+                    {bodyColumnContent}
+                </Paper>
+            </Grid>
+        </>
+    );
+};
+
+// -- renders a cell of type 'markdown'
+const JupyterNotebookMarkdownCell: React.FC<JupyterNotebookCellProps> = ({ cell, execution_count, language }) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={''}
+            bodyColumnContent={<ReactMarkdown children={sourceLines(cell.source)} />}
+            bodyColumnStyle={{ padding: '0px 8px 0px 8px', margin: '0px 0px', border: '1px solid #AAA' }}
+        />
+    );
+};
+
+// -- renders a cell of type 'heading'
+const JupyterNotebookHeadingCell: React.FC<JupyterNotebookCellProps> = ({ cell, execution_count, language }) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={''}
+            bodyColumnContent={
+                <Container
+                    style={{
+                        marginBottom: '0px',
+                        boxShadow: 'none',
+                    }}
+                >
+                    <Heading variant="h1">{cell.source[0]}</Heading>
+                </Container>
+            }
+            bodyColumnStyle={{ padding: '0px 8px 0px 8px', margin: '0px 0px', border: '1px solid #AAA' }}
+        />
+    );
+};
+
+// -- renders a cell of type 'stream'
+const JupyterNotebookOutputStreamCell: React.FC<JupyterNotebookOutputCellProps> = ({
+    cell,
+    execution_count,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`Out [${execution_count}]`}
+            bodyColumnContent={
+                <ReactMarkdown children={'**[' + output.name + ']**\n```\n ' + sourceLines(output.text) + '```'} />
+            }
+            bodyColumnStyle={{
+                padding: '0px 8px 0px 8px',
+                margin: '0px 0px',
+                backgroundColor: output.name === 'stderr' ? '#FDD' : '#DFD',
+                border: '1px solid #AAA',
+            }}
+        />
+    );
+};
+
+// -- renders a cell of type 'error'
+const JupyterNotebookOutputErrorCell: React.FC<JupyterNotebookOutputCellProps> = ({
+    cell,
+    execution_count,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`In [${execution_count}]`}
+            bodyColumnContent={
+                <ReactMarkdown
+                    children={
+                        '**[' +
+                        output.evalue +
+                        ']**\n```\n ' +
+                        sourceLines(output.traceback).replace(
+                            /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g,
```

Comment:
Please could you add a comment (or even better a unit test) for this regex? :) (Might just be me that can't read it though 😂 )

Reply:
will add comment - it's to strip ansi control codes from err cell entries (for aesthetic reasons)